### ISSUE TITLE
Custom methods cannot access parent scope

### ISSUE THREAD

**Issue by DonGissel** [sveltejs/svelte.technology#21]
Let's say I'm building a repeatable component that ties to some kind of data from a server. I want to create a button to update the data on the server (let's say I'll increase a counter by one), and while the `fetch()` is running I would like to set a loading-state on my component. However, once I've bound my custom method to an element, I can't seem to access the scope from within the method.

A quick example:

`<button type="button" on:click="increaseCount(task.id)">`

Will then fire:

```
methods: {
    increaseCount: (id) => {
        this.set({ loading: true });
        fetch(`${settings.endPoints.task}/${id}/increase`, { method: 'POST' })
           .then(() => {
                this.set({ loading: false });
            });
    }
}
```

All this is defined in my `.svelte`-file in the `<script>`-block. However, when the event fires, `this` is undefined (so `this.set()` doesn't work), `set` doesn't exist as a global variable (go figure), and I can't seem to reference my scope any other way. I tried passing `this` on from the on:click and assigning it to a variable, but that just pointed me to the button, not the component. I even tried defining my component as a named variable first, like so:

```
const taskComponent = {
    // code here, in which I try to solve the problem by
    // pointing to taskComponent.set() in my method
};
export default taskComponent;
```

... but that just gave me a nasty forEach-error in the compiler (using `rollup-plugin-svelte`).

So, to summarize: I'd like very much to be able to access my scope/component from within my custom methods in order to do stuff like this, as well as maybe update data dynamically with whatever my query might return, so the local data is looking right until the "master query" fires and updates everything properly.

**Comment by DonGissel** [comment]
Ahem. I see now that this is the repo for the website. I'll reopen elsewhere. ;-) Sorry!
